hoss/dimension_utilities.py

```diff
@@ -73,38 +113,133 @@ def prefetch_dimension_variables(
-
     return required_dimensions_nc4
 
 
+def is_variable_one_dimensional(
+    prefetch_dataset: Dataset, dimension_variable: VariableFromDmr
+) -> bool:
+    """Check if a dimension variable is 1D"""
+    dimensions_array = prefetch_dataset[dimension_variable.full_name_path][:]
+    return dimensions_array.ndim == 1
+
+
+def update_dimension_variables(
+    prefetch_dataset: Dataset,
+    required_dimensions: Set[str],
+    varinfo: VarInfoFromDmr,
+) -> Dict[str, ndarray]:
+    """Augment a NetCDF4 file with artificial 1D dimensions variable for each
+    2D dimension variable"
+
+    For each dimension variable:
+    (1) Check if the dimension variable is 1D.
+    (2) If it is not 1D and is 2D create a dimensions array from
+        within the `write_1D_dimensions`
+        function.
+    (3) Then write the 1D dimensions variable to the NetCDF4 URL.
+
+    """
+    for dimension_name in required_dimensions:
+        dimension_variable = varinfo.get_variable(dimension_name)
+        if not is_variable_one_dimensional(prefetch_dataset, dimension_variable):
+            col_size = prefetch_dataset[dimension_variable.full_name_path][:].shape[0]
+            row_size = prefetch_dataset[dimension_variable.full_name_path][:].shape[1]
+            crs = get_variable_crs(dimension_name, varinfo)
+
+    geo_grid_corners = get_geo_grid_corners(
+        prefetch_dataset, required_dimensions, varinfo
+    )
+
+    x_y_extents = get_x_y_extents_from_geographic_points(geo_grid_corners, crs)
+
+    # get grid size and resolution
+    x_min = x_y_extents['x_min']
+    x_max = x_y_extents['x_max']
+    y_min = x_y_extents['y_min']
+    y_max = x_y_extents['y_max']
+    x_resolution = (x_max - x_min) / row_size
+    y_resolution = (y_max - y_min) / col_size
+
+    # create the xy dim scales
+    x_dim = np.arange(x_min, x_max, x_resolution)
+
+    # ascending versus descending..should be based on the coordinate grid
+    y_dim = np.arange(y_max, y_min, -y_resolution)
+    return {'projected_y': y_dim, 'projected_x': x_dim}
+
+
+def get_geo_grid_corners(
+    prefetch_dataset: Dataset,
+    required_dimensions: Set[str],
+    varinfo: VarInfoFromDmr,
+) -> list[Tuple[float]]:
+    """
+    This method is used to return the lat lon corners from a 2D
+    coordinate dataset. This does a check for values below -180
+    which could be fill values. Does not check if there are fill
+    values in the corner points to go down to the next row and col
+
+    """
+    for dimension_name in required_dimensions:
+        dimension_variable = varinfo.get_variable(dimension_name)
+        if dimension_variable.is_latitude():
+            lat_arr = prefetch_dataset[dimension_variable.full_name_path][:]
+        elif dimension_variable.is_longitude():
+            lon_arr = prefetch_dataset[dimension_variable.full_name_path][:]
+
+    # skip fill values when calculating min values
+    # topleft =  minlon, maxlat
+    # bottomright = maxlon, minlat
```

Comment:
I don't think we can make that assumption in general. It might be true of the SMAP L3 collections, but HOSS is meant to be used across the enterprise.

Reply:
corrected that - https://github.com/nasa/harmony-opendap-subsetter/pull/15/commits/f07b544fd6dca4e8e9af53690a645c54124c9db7

Reply:
I'm going to assume this is handled in a later commit with a refactor of the f07b544.